compiler/src/iree/compiler/Codegen/SPIRV/SPIRVTile.cpp

```diff
@@ -46,11 +46,19 @@ namespace iree_compiler {
 /// Collects computation ops which we will use as anchor to tile and fuse.
 static FailureOr<IREE::Codegen::LoweringConfigAttr> collectComputeOps(
     func::FuncOp funcOp, SmallVectorImpl<Operation *> &computeOps) {
-  // If there are `scf.if` ops, we have both a fast and slow paths for
-  // padding handling. Then we need to scan both regions to discover such
-  // computation ops so that we can tile and fuse both regions.
+  // If there are `scf.if` ops which have linalg ops, we have both a fast and
+  // slow paths for padding handling. Then we need to scan both regions to
+  // discover such computation ops so that we can tile and fuse both regions.
   SmallVector<scf::IfOp, 1> ifOps;
-  funcOp.walk([&ifOps](scf::IfOp ifOp) { ifOps.push_back(ifOp); });
+  funcOp.walk<WalkOrder::PreOrder>([&ifOps](Operation *op) -> WalkResult {
+    if (isa<linalg::LinalgOp, TilingInterface>(op)) {
```

Comment:
Could you just check in the walk if the parent is a `LinalgOp` ?

Reply:
+1

Reply:
Fixed, thanks.

Reply:
I think the suggestion was to do something like

```c++
funcOp.walk<PreOrder>([&ifOps](scf::IfOp ifOp) {
  if (ifOp->getParentOfType<linalg::LinalgOp>()) // skip
  else // push and advance
});
```

Reply:
Thank you for the clarification. Now I think it has been fixed accordingly.